test/unit/kubernetes-deploy/kubernetes_resource/horizontal_pod_autoscaler_test.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class HorizontalPodAutoscalerTest < KubernetesDeploy::TestCase
+  include ResourceCacheTestHelper
+
+  # We can't get integration coverage for HPA right now because the metrics server just isn't reliable enough on our CI
+  def test_hpa_is_whitelisted_for_pruning
+    KubernetesDeploy::Kubectl.any_instance.expects("run")
+      .with("get", "CustomResourceDefinition", "-a", "--output=json", attempts: 5)
+      .returns(['{ "items": [] }', "", SystemExit.new(0)])
+    task = KubernetesDeploy::DeployTask.new(namespace: 'test', context: KubeclientHelper::TEST_CONTEXT,
+      current_sha: 'foo', template_dir: '', logger: logger)
+    assert(task.prune_whitelist.one? { |whitelisted_type| whitelisted_type.include?("HorizontalPodAutoscaler") })
+  end
+
+  def test_hpa_succeeds_when_scaling_is_active
+    conditions = [{
+      "lastTransitionTime" => 5.seconds.ago,
+      "message" => "the HPA was able to successfully calculate a replica count from cpu "\
+        "resource utilization (percentage of request)",
+      "reason" => "ValidMetricFound",
+      "status" => "True",
+      "type" => "ScalingActive",
+    }]
+    hpa = build_synced_hpa(build_hpa_template(conditions: conditions))
+    assert(hpa.deploy_succeeded?)
+    refute(hpa.deploy_failed?)
+    assert_equal("Configured", hpa.status)
+  end
+
+  def test_hpa_succeeds_if_scaling_is_explicitly_disabled
```

Comment:
With the changes you made to the HPA class. I think we could combine this test and the pruning one as the `test_hpa_can_be_pruned` integration test if we sent the deployment to `0` replicas.

Reply:
Good point! I gave this a try though, and seemingly the HPA controller is really slow. Even the `ScalingDisabled` condition reliably takes at least 30s to appear when I run the test. I'm not sure covering that condition is worth that much time 😞

Reply:
Wonder if that 30 seconds is actually a hard coded value somewhere. But since the integration tests run in parallel, I'd trade the machine time for more fidelity.

Reply:
It's more fidelity only on the edge case where the've got an HPA that is disabled. If it were on a primary success or failure case I'd agree, but not 30s for a ScalingDisabled test. It's run in parallel yes, but we can't control at which point in the run this particular test will start.